tests/integration/conftest.py

```diff
@@ -219,3 +227,36 @@ def restore_old_state(old_state):
                     )
     if len(desire_state["interfaces"]):
         libnmstate.apply(desire_state, verify_change=False)
+
+
+def _setup_state_dump():
+    shutil.rmtree(STATES_DUMP_DIR, ignore_errors=True)
+    libnmstate.apply = _libnmstate_dump_and_apply
+
+
+def _libnmstate_dump_and_apply(
```

Comment:
Add an option to run the checker directly instead of dumping files 
```python
def pytest_collection_modifyitems(config, items):
    if config.getoption("--go-checker"):
        libnmstate.apply = _custom_apply_with_go_checker
```